packages/nimble-components/src/table-column/enum-base/index.ts

```diff
@@ -0,0 +1,123 @@
+import {
+    attr,
+    Notifier,
+    Observable,
+    observable,
+    Subscriber
+} from '@microsoft/fast-element';
+import type {
+    TableStringField,
+    TableBooleanField,
+    TableNumberField
+} from '../../table/types';
+import { TableColumn } from '../base';
+import { Mapping } from '../../mapping/base';
+import type { MappingKeyType } from './types';
+import type { MappingConfig } from './models/mapping-config';
+import type { MappingKey } from '../../mapping/base/types';
+import { resolveKeyWithType } from './models/mapping-key-resolver';
+
+export type TableColumnEnumCellRecord =
+    | TableStringField<'value'>
+    | TableBooleanField<'value'>
+    | TableNumberField<'value'>;
+export type MappingConfigs = Map<MappingKey, MappingConfig>;
+export interface TableColumnEnumColumnConfig {
+    mappingConfigs: MappingConfigs;
+}
+
+/**
+ * Base class for table columns that map values to content (e.g. nimble-table-column-enum-text and nimble-table-column-icon)
+ */
+export abstract class TableColumnEnumBase<
+    TColumnConfig extends TableColumnEnumColumnConfig,
+    TEnumValidator
+>
+    extends TableColumn<TColumnConfig>
+    implements Subscriber {
+    /** @internal */
```

Comment:
nit: Can we move `internal` members below non-internal public members?

Reply:
Actually, we need this member to come first in the order. The members get initialized in order, and in the `*Changed()` handlers, we want the validator to be initialized already. I'll add a comment to that effect.